JSAPI.md

```diff
@@ -0,0 +1,62 @@
+# Draft of the JavaScript API
+
+This is currently a work in progress, expect the API to change significantly.
+Specifically, the APIs returning strings as results of inspection should return
+structured data instead for ease of use.
+
+```js
+class LLNode {
+  /**
+   * @param {string} dump path to the coredump
+   * @param {string} executable path to the node executable
+   * @returns {LLNode} an LLNode instance
+   */
+  static fromCoredump(dump, executable) {}
```

Comment:
```js
class LLNode {
  constructor(dump, executable){ }
  
  // LLDB load coredump and executable
  loadCore()
}
```

Is this form better?

Reply:
@hyj1991 I'd like to avoid overloading the constructor, since loading from a core dump should not be the only way to start an llnode instance, we may want to be able to attach to a process or launch a process via the API in the future.

Reply:
@joyeecheung  looking forward to it